webapp/channels/src/components/advanced_text_editor/advanced_text_editor.tsx

```diff
@@ -409,6 +410,13 @@ const AdvancedTextEditor = ({
         }
     }, [showPreview]);
 
+    // Add useEffect to focus textbox when selectedPostFocussedAt changes
+    useEffect(() => {
+        if (selectedPostFocussedAt) {
+            focusTextbox();
+        }
+    }, [selectedPostFocussedAt, focusTextbox]);
```

Comment:
Here, you are saying that whenever `focusTextbox` changes (while selectedPostFocusedAt is set) also focus the textbox. Indirectly, that implies that anytime textboxRef, channelId, isRHS or canPost (the props for useTextboxFocus) changes it will focus the textbox. Most of that doesn't seem too dangerous as of today (isRHS is not going to change, if channel id changes it probably will need to get the focus either way, textboxRef is a ref, so it will not change) but `canPost` can lead you to weird situations if it not were because of the `selectedPostFocussedAt` check. If we didn't have that (or it misbehaves keeping the value longer than what is intended) you could be writing in the RHS while the center pane has a read only channel. If you get permissions to write on that channel, canPost wil change, focusTextbox may potentially change (I don't remember the implementation, but it can change since a different prop is passed), and suddenly the center pane will steal the focus.

Reply:
ok, I ’ll update the code to remove focusTextbox from the effect’s dependencies, making sure we only focus when selectedPostFocussedAt is updated and not because to unrelated prop changes.